backend/app/usecase/search/filter.go

```diff
@@ -10,9 +10,14 @@ const (
 	User
 )
 
+// OrderedResource represents a type of searchable objects together with the order.
+type OrderedResource struct {
+	Resource Resource
+	Order    order.By
+}
```

Comment:
I like the idea of combining them into a single struct. However, it does look weird especially when we are adding 1 more attribute in the future. Can we add `func NewFilter() (Filter, error)` and `(f *F) IsValid() bool` and return error in search function instead?

Reply:
Okay, let me implement your suggestion then we can compare all three alternatives